File: Python/HackerRank/beautifulDays.py

def beautifulDays(i, j, k):
    c = 0
    for i in range(i, j + 1):
        if (i - int(str(i)[::-1])) % k == 0:
            c += 1
    return c

def main():
    i, j, k = map(int, input().split())
    print(beautifulDays(i, j, k))

if __name__ == '__main__':
    main()

File: problems/hackerrank/math/beautifulDays/beautifulDays.go
package math



// BeautifulDays counts the number of beautiful days in a range.
func BeautifulDays(i, j, k int) int {
	count := 0
	for d := i; d <= j; d++ {
		diff := d - reverseInt(d)
		if diff < 0 {
			diff = -diff
		}
		if diff%k == 0 {
			count++
		}
	}
	return count
}

func reverseInt(n int) int {
	res := 0
	for n > 0 {
		res = res*10 + n%10
		n /= 10
	}
	return res
}
